lib/webmushra/pages/RankingPage.js

```diff
@@ -0,0 +1,469 @@
+
+  var tdConditionNamesScale = $("<td id='conditionNameScale'></td>");
+  trConditionNames.append(tdConditionNamesScale);
+
+  var conditions = this.mushraAudioControl.getConditions();
+  var i;
+  var idx = 0;
+  for (i = 0; i < conditions.length; ++i) {
+    var str = "";
+    if (this.pageConfig.showConditionNames === true) {
+      str = "<br/>" + conditions[i].id;
+    }
+    td = $("<td>" + this.pageManager.getLocalizer().getFragment(this.language, 'cond') + (idx + 1) + str + "</td>");
+    trConditionNames.append(td);
+    idx = idx + 1;
+  }
+
+  this.numCond = idx;
+  this.score = new Array(this.conditions.length).fill(0);
+  this.rank = 1;
+  this.itemEliminated = -1;
+
+  var trConditionPlay = $("<tr></tr>");
+  tableMushra.append(trConditionPlay);
+
+  var tdConditionPlayReference = $("<td></td>");
+  trConditionPlay.append(tdConditionPlayReference);
+
+  var tdConditionPlayScale = $("<td></td>");
+  trConditionPlay.append(tdConditionPlayScale);
+
+  for (i = 0; i < conditions.length; ++i) {
+    td = $("<td></td>");
+    var buttonPlay = $("<button data-role='button' class='center audioControlElement' onclick='" + this.pageManager.getPageVariableName(this) + ".btnCallbackCondition(" + i + ");'>" + this.pageManager.getLocalizer().getFragment(this.language, 'playButton') + "</button>");
+    buttonPlay.attr("id", "buttonConditions" + i);
+    td.append(buttonPlay);
+    trConditionPlay.append(td);
+  }
+
+  // ratings
+  var trConditionRatings = $("<tr id='tr_ConditionRatings'></tr>");
+  tableMushra.append(trConditionRatings);
+
+  var tdConditionRatingsReference = $("<td id='refCanvas'></td>");
+  trConditionRatings.append(tdConditionRatingsReference);
+
+  var tdConditionRatingsScale = $("<td></td>");
+  trConditionRatings.append(tdConditionRatingsScale);
+
+  for (i = 0; i < conditions.length; ++i) {
+    td = $("<td></td>");
+    var buttonElim = $("<button data-role='button' onclick='" + this.pageManager.getPageVariableName(this) + ".btnCallbackElimination(" + i + ");'>" + this.pageManager.getLocalizer().getFragment(this.language, 'eliminateButton') + "</button>");
+    buttonElim.attr("id", "buttonElim" + i);
+    td.append(buttonElim);
+    trConditionRatings.append(td);
+  }
+
+  if (this.pageConfig.showResetButton === true){
+    var buttonReset = $("<td><button data-role='button' onclick='" + this.pageManager.getPageVariableName(this) + ".btnCallbackReset();'>" + this.pageManager.getLocalizer().getFragment(this.language, 'resetButton') + "</button></td>");
+    trConditionRatings.append(buttonReset);
+  }
+
+
+  this.macic = new MushraAudioControlInputController(this.mushraAudioControl, this.pageConfig.enableLooping);
+  this.macic.bind();
+
+  this.waveformVisualizer = new WaveformVisualizer(this.pageManager.getPageVariableName(this) + ".waveformVisualizer", tdRight, this.conditions[0], this.pageConfig.showWaveform, this.pageConfig.enableLooping, this.mushraAudioControl);
```

Comment:
isn't it an arbitrary decision to select `this.conditions[0]` as the condition that is used to render the waveform? I feel like overall it would be better to use the reference for this and therefore give experiments control over which condition or even the reference is used?

Reply:
@kishorkl can you comment on this one please?

Reply:
@faroit you are right that using this.conditions[0] to plot the waveforms is arbitrary. I would need to reread the paper to check what they suggest in terms of using the reference. I will get back on this in two weeks.

Reply:
Great. Thanks

Reply:
hi, do you have some update on this one?

Reply:
@kishorkl do you have time to finish this?